test/mpi/dtpools/include/dtpools_internal.h

```diff
@@ -121,9 +121,94 @@ typedef enum {
+            case DTPI_OBJ_LAYOUT_LARGE_BLK__INDEXED:                    \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT__INDEXED:                    \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__INDEXED:               \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__INDEXED:               \
+                obj_type = DTPI_OBJ_TYPE__##pool_type##INDEXED;         \
+                break;                                                  \
+            case DTPI_OBJ_LAYOUT_SIMPLE__HINDEXED:                      \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK__HINDEXED:                   \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT__HINDEXED:                   \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__HINDEXED:              \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__HINDEXED:              \
+                obj_type = DTPI_OBJ_TYPE__##pool_type##HINDEXED;        \
+                break;                                                  \
+            case DTPI_OBJ_LAYOUT_SIMPLE__BLOCK_INDEXED:                 \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK__BLOCK_INDEXED:              \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT__BLOCK_INDEXED:              \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__BLOCK_INDEXED:         \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__BLOCK_INDEXED:         \
+                obj_type = DTPI_OBJ_TYPE__##pool_type##BLOCK_INDEXED;   \
+                break;                                                  \
+            case DTPI_OBJ_LAYOUT_SIMPLE__BLOCK_HINDEXED:                \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK__BLOCK_HINDEXED:             \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT__BLOCK_HINDEXED:             \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__BLOCK_HINDEXED:        \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__BLOCK_HINDEXED:        \
+                obj_type = DTPI_OBJ_TYPE__##pool_type##BLOCK_HINDEXED;  \
+                break;                                                  \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK__SUBARRAY_C:                 \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT__SUBARRAY_C:                 \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__SUBARRAY_C:            \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__SUBARRAY_C:            \
+                obj_type = DTPI_OBJ_TYPE__##pool_type##SUBARRAY_C;      \
+                break;                                                  \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK__SUBARRAY_F:                 \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT__SUBARRAY_F:                 \
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__SUBARRAY_F:            \
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__SUBARRAY_F:            \
+                obj_type = DTPI_OBJ_TYPE__##pool_type##SUBARRAY_F;      \
+                break;                                                  \
+            default:;                                                   \
```

Comment:
How about having a (failing) assertion here for the default case?

Reply:
I will fix this